tutorials/Inference-PyTorch-Bert-Model-for-High-Performance-in-ONNX-Runtime.ipynb

```diff
@@ -0,0 +1,344 @@
+    "    symbolic_names = {0: 'batch_size', 1: 'max_seq_len'}\n",
+    "    torch.onnx.export(model,                                            # model being run\n",
+    "                      (inputs['input_ids'],                             # model input (or a tuple for multiple inputs)\n",
+    "                       inputs['attention_mask'], \n",
+    "                       inputs['token_type_ids']), \n",
+    "                      output_model_path,                                # where to save the model (can be a file or file-like object)\n",
+    "                      opset_version=11,                                 # the ONNX version to export the model to\n",
+    "                      do_constant_folding=True,                         # whether to execute constant folding for optimization\n",
+    "                      input_names=['input_ids',                         # the model's input names\n",
+    "                                   'input_mask', \n",
+    "                                   'segment_ids'],\n",
+    "                      output_names=['start', 'end'],                    # the model's output names\n",
+    "                      dynamic_axes={'input_ids': symbolic_names,        # variable length axes\n",
+    "                                    'input_mask' : symbolic_names,\n",
+    "                                    'segment_ids' : symbolic_names,\n",
+    "                                    'start' : symbolic_names,\n",
+    "                                    'end' : symbolic_names})\n",
+    "    print(\"Model exported at \", output_model_path)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 3. Inference the Exported Model with ONNX Runtime ##\n",
+    "\n",
+    "#### Install ONNX Runtime\n",
+    "Install ONNX Runtime if you haven't done so already. \n",
+    "\n",
+    "Install `onnxruntime` to use CPU features, or `onnxruntime-gpu` to use GPU. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "ONNXRUNTIME = 'onnxruntime'\n",
+    "# Install ONNX Runtime\n",
+    "if torch.cuda.is_available():\n",
+    "    ## Install onnxruntime-gpu if cuda is available\n",
+    "    ONNXRUNTIME = 'onnxruntime-gpu'\n",
+    "\n",
+    "import sys\n",
+    "!{sys.executable} -m pip install -U $ONNXRUNTIME"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now we are ready to inference the model with ONNX Runtime"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import onnxruntime as rt  \n",
+    "\n",
+    "sess_options = rt.SessionOptions()\n",
+    "\n",
+    "# Set graph optimization level to ORT_ENABLE_EXTENDED to enable bert optimization.\n",
+    "sess_options.graph_optimization_level = rt.GraphOptimizationLevel.ORT_ENABLE_EXTENDED\n",
```

Comment:
I thought the default setting is ALL, so why reduce the optimization level?

Reply:
In `onnxruntime` v1.1, the default graph optimization level is `BASIC`. Therefore we need to manually increase the level to `EXTENDED` to enable the BERT optimization. However, in the upcoming patch release, the default level will be `ALL` so there will be no need to set optimization level. This line will be removed with the patch release.

Reply:
In onnxruntime 1.1, the default graph optimization level is BASIC. So we need to manually increase the level to EXTENDED to enable the bert optimization. However in the upcoming patch release, the default level will be ALL, so no need to set optimization level. This line will be removed with the patch release.